### PR TITLE
refactor(mandates): refactor mandates to check for misleading error codes in mandates

### PR DESCRIPTION
**Pull request by BallaNitesh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Refactored code while checking for misleading error codes.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/errors/api_error_response.rs [hunk 1]
```diff
@@ -148,6 +148,10 @@ pub enum ApiErrorResponse {
     MandateUpdateFailed,
     #[error(error_type = ErrorType::ObjectNotFound, code = "HE_02", message = "API Key does not exist in our records")]
     ApiKeyNotFound,
+    #[error(error_type = ErrorType::ValidationError, code = "HE_03", message = "Unable to find the mandate in our records.")]
```
Comment (NishantJoshi00) [inline crates/router/src/core/errors/api_error_response.rs:151]: The error variant, and the message doesn't match can we try and match them?

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
other than that looks good to me

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
Other than that, LGTM

File: crates/router/src/core/errors/api_error_response.rs (hunk 1 above)
Comment (BallaNitesh) [reply crates/router/src/core/errors/api_error_response.rs:151]: Making this "invalid mandate id passed from connector."

**Review by inventvenkat — APPROVED** [review]

**Comment by inventvenkat** [comment]
Can you add test records here in the PR

**Review by inventvenkat — DISMISSED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Review by inventvenkat — APPROVED** [review]

**Pull request merged**